magicblock-committor-service/src/intent_executor/single_stage_executor.rs

```diff
@@ -0,0 +1,190 @@
+            // Note that this not necessarily will pass at the end due to the same reason
+
+            // SAFETY: is_some() checked prior
+            let committed_pubkeys = committed_pubkeys.unwrap();
+            let (commit_strategy, finalize_strategy, cleanup) =
+                self.handle_cpi_limit_error(last_transaction_strategy);
+            junk.push(cleanup);
+            self.two_stage_execution_flow(
+                &committed_pubkeys,
+                commit_strategy,
+                finalize_strategy,
+                persister,
+            )
+            .await
+        } else {
+            junk.push(last_transaction_strategy);
+            let err = IntentExecutorError::from_strategy_execution_error(
+                execution_err,
+                |internal_err| {
+                    let signature = internal_err.signature();
+                    IntentExecutorError::FailedToFinalizeError {
+                        err: internal_err,
+                        commit_signature: signature,
+                        finalize_signature: signature,
+                    }
+                },
+            );
+
+            Err(err)
+        }
+    }
+
+    pub async fn patch_strategy(
+        &self,
+        err: &TransactionStrategyExecutionError,
+        transaction_strategy: &mut TransactionStrategy,
+        base_intent: &ScheduledBaseIntent,
+    ) -> IntentExecutorResult<ControlFlow<(), TransactionStrategy>> {
+        let Some(committed_pubkeys) = base_intent.get_committed_pubkeys()
+        else {
+            // No patching is applicable if intent doesn't commit accounts
+            return Ok(ControlFlow::Break(()));
+        };
+
+        match err {
+            TransactionStrategyExecutionError::ActionsError => {
+                // Here we patch strategy for it to be retried in next iteration
+                // & we also record data that has to be cleaned up after patch
+                let to_cleanup =
+                    self.handle_actions_error(transaction_strategy);
+                Ok(ControlFlow::Continue(to_cleanup))
+            }
+            TransactionStrategyExecutionError::CommitIDError => {
+                // Here we patch strategy for it to be retried in next iteration
+                // & we also record data that has to be cleaned up after patch
+                let to_cleanup = self
+                    .handle_commit_id_error(
+                        &committed_pubkeys,
+                        transaction_strategy,
+                    )
+                    .await?;
+                Ok(ControlFlow::Continue(to_cleanup))
+            }
+            TransactionStrategyExecutionError::CpiLimitError => {
+                // Can't be handled in scope of single stage execution
+                // We signal flow break
+                Ok(ControlFlow::Break(()))
+            }
+            TransactionStrategyExecutionError::InternalError(_) => {
```

Comment:
Again loosing context here. We'll never know why it failed executing.
Basically anytime we just `_` an error context we should reconsider.